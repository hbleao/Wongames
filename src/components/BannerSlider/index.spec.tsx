import React from 'react';
import { screen } from '@testing-library/react';

import { BannerSlider } from '.';
import { renderWithTheme } from 'utils/tests/renderWithTheme';

const items = [
  {
    image: 'https://source.unsplash.com/user/willianjusten/1042x580',
    title: 'Defy death 1',
    subtitle: '<p>Play the new <strong>Crashland</strong> season</p>',
    buttonLabel: 'Buy now',
    buttonLink: '/games/defy-death'
  },
  {
    image: 'https://source.unsplash.com/user/willianjusten/1042x580',
    title: 'Defy death 2',
    subtitle: '<p>Play the new <strong>Crashland</strong> season</p>',
    buttonLabel: 'Buy now',
    buttonLink: '/games/defy-death'
  }
];

const makeSut = () => {
  const sut = renderWithTheme(<BannerSlider items={items} />);

  return {
    sut
  };
};

describe('BannerSlider', () => {
  it('should be render a BannerSlider', () => {
    makeSut();

    const element = document.querySelector('.slick-vertical');

    expect(element).toBeInTheDocument();
  });

  it('should render with 1 active item', () => {
    makeSut();

    const banners = document.querySelectorAll('.slick-slide');
    const activeBanner = document.querySelectorAll('li.slick-active');
    const firstBannerTitle = screen.getByRole('heading', {
      name: 'Defy death 1',
      hidden: false
    });
    const secondBannerTitle = screen.getByRole('heading', {
      name: 'Defy death 2',
      hidden: true
    });

    expect(banners).toHaveLength(2);
    expect(activeBanner).toHaveLength(1);
    expect(firstBannerTitle).toBeInTheDocument();
    expect(secondBannerTitle).toBeInTheDocument();
  });

  it('should render 2 dots', () => {
    makeSut();

    const dotItems = document.querySelectorAll('ul.slick-dots > li');

    expect(dotItems).toHaveLength(2);
  });
});
