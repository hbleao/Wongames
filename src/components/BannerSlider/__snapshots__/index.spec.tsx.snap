// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/BannerSlider should be render a BannerSlider 1`] = `
.c10 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #FAFAFA;
  background-color: #F231A5;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 0 2.4rem;
  right: -2rem;
}

.c10::before {
  border-left-color: #b20b6f;
  border-top-color: #b20b6f;
}

.c10::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

.c10::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c3 {
  position: relative;
}

.c4 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(https://source.unsplash.com/user/willianjusten/1042x580);
  background-size: cover;
}

.c5 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c6 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c7 {
  font-size: 1.4rem;
  font-weight: 400;
  color: #FAFAFA;
  margin-bottom: 1.6rem;
}

.c8 {
  color: #FAFAFA;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  background-image: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c8:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c1 .slick-slider {
  position: relative;
  display: block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.c1 .slick-list {
  position: relative;
  display: block;
  overflow: hidden;
  margin: 0;
  padding: 0;
}

.c1 .slick-list:focus {
  outline: none;
}

.c1 .slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.c1 .slick-slider .slick-track,
.c1 .slick-slider .slick-list {
  -webkit-transform: translate3d(0,0,0);
  -moz-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  -o-transform: translate3d(0,0,0);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c1 .slick-track {
  position: relative;
  top: 0;
  left: 0;
  display: block;
}

.c1 .slick-track:before,
.c1 .slick-track:after {
  display: table;
  content: '';
}

.c1 .slick-track:after {
  clear: both;
}

.c1 .slick-loading .slick-track {
  visibility: hidden;
}

.c1 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

.c1 [dir='rtl'] .slick-slide {
  float: right;
}

.c1 .slick-slide img {
  display: block;
}

.c1 .slick-slide.slick-loading img {
  display: none;
}

.c1 .slick-slide.dragging img {
  pointer-events: none;
}

.c1 .slick-initialized .slick-slide {
  display: block;
}

.c1 .slick-loading .slick-slide {
  visibility: hidden;
}

.c1 .slick-vertical .slick-slide {
  display: block;
  height: auto;
  border: 1px solid transparent;
}

.c1 .slick-arrow.slick-hidden {
  display: none;
}

.c0 {
  overflow: hidden;
}

.c0 .slick-dots {
  list-style: none;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 2.4rem;
}

.c0 .slick-dots li {
  background-color: #FAFAFA;
  width: 1.2rem;
  height: 1.2rem;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 0.8rem;
  cursor: pointer;
}

.c0 .slick-dots li.slick-active {
  background: #F231A5;
}

.c0 .slick-dots button {
  opacity: 0;
  width: 1.2rem;
  height: 1.2rem;
  cursor: pointer;
}

@media (max-width:1170px) {
  .c3 .c9 {
    right: 0;
  }

  .c3 .c9::before {
    display: none;
  }
}

@media (min-width:768px) {
  .c4 {
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c5 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c7 {
    font-size: 1.8rem;
  }
}

@media (min-width:1170px) {
  .c0 .c2 {
    max-width: 104rem;
    margin: 0 auto;
  }

  .c0 .slick-dots {
    position: absolute;
    top: 0;
    right: 0;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    height: 100%;
    margin: 0;
  }

  .c0 .slick-dots li {
    margin: 0.8rem 0;
  }
}

<div>
  <section
    class="c0"
  >
    <section
      class="c1"
    >
      <div
        class="slick-slider slick-vertical slick-initialized"
        dir="ltr"
      >
        <div
          class="slick-list"
          style="height: 0px;"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="c2 c3"
                >
                  <div
                    aria-label="Defy death 1"
                    class="c4"
                    role="img"
                    src="https://source.unsplash.com/user/willianjusten/1042x580"
                  />
                  <div
                    class="c5"
                  >
                    <h2
                      class="c6"
                    >
                      Defy death 1
                    </h2>
                    <h3
                      class="c7"
                    >
                      <p>
                        Play the new 
                        <strong>
                          Crashland
                        </strong>
                         season
                      </p>
                    </h3>
                    <a
                      class="c8"
                      href="/games/defy-death"
                    >
                      <span>
                        Buy now
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="c2 c3"
                >
                  <div
                    class="c9 c10"
                    color="primary"
                  >
                    20% OFF
                  </div>
                  <div
                    aria-label="Defy death 2"
                    class="c4"
                    role="img"
                    src="https://source.unsplash.com/user/willianjusten/1042x580"
                  />
                  <div
                    class="c5"
                  >
                    <h2
                      class="c6"
                    >
                      Defy death 2
                    </h2>
                    <h3
                      class="c7"
                    >
                      <p>
                        Play the new 
                        <strong>
                          Crashland
                        </strong>
                         season
                      </p>
                    </h3>
                    <a
                      class="c8"
                      href="/games/defy-death"
                    >
                      <span>
                        Buy now
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="2"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="c2 c3"
                >
                  <div
                    aria-label="Defy death 3"
                    class="c4"
                    role="img"
                    src="https://source.unsplash.com/user/willianjusten/1042x580"
                  />
                  <div
                    class="c5"
                  >
                    <h2
                      class="c6"
                    >
                      Defy death 3
                    </h2>
                    <h3
                      class="c7"
                    >
                      <p>
                        Play the new 
                        <strong>
                          Crashland
                        </strong>
                         season
                      </p>
                    </h3>
                    <a
                      class="c8"
                      href="/games/defy-death"
                    >
                      <span>
                        Buy now
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <ul
          class="slick-dots"
          style="display: block;"
        >
          <li
            class="slick-active"
          >
            <button>
              1
            </button>
          </li>
          <li
            class=""
          >
            <button>
              2
            </button>
          </li>
          <li
            class=""
          >
            <button>
              3
            </button>
          </li>
        </ul>
      </div>
    </section>
  </section>
</div>
`;
